### DISCUSSION TITLE
triggering midi event every 2 seconds without any action

### DISCUSSION THREAD

**Discussion by mdnkndt** [rsjaffe/MIDI2LR#727] [Problems]
Hello,
first, thank you very much for this app. Using midi to edit photos in LR is a great idea.
I recently bought a streamdeck plus. I use this midi plugin to send midi CC to midi2LR ([https://trevligaspel.se/streamdeck/midi/index.html](url)).
When LR is launched, everything's fine, midi2LR is launched too and well connected.
Until I turn a dial button on my streamdeck. midi CC is well transmitted to LR and value of selected parameter changes.
But from this moment, even if I don't touch anything, midi2LR info field under the CC list blinks green every 2 seconds displaying 1:CC40 [46], no matter which dial I first turned on the streamdeck.
I can't find midi CC received or sent by midi2LR in the app logs to try to find the source of the bug.
It's annoying, because parameter's value is constantly displayed on the photo I'm currently editing.
Am I clear enough ?
Logs and xml can be downloaded here : [https://drive.google.com/file/d/1jSu9tVc9YqJcoHgRrPhY5QxDbPo08kUd/view?usp=share_link](url)

Thank you

Mathieu

**Comment by dkulik51** [comment]
This is also happening to me. I recently upgraded from a 2008 Mac Pro to a 2021 MacBook Pro M1 Max, so was able to upgrade from my old version of Lightroom to LR Classic 12.1. I had been running MIDI2LR version 2.2.0.0 with the old Lightroom, and when I upgraded to version 5.4.0.0 this started happening. From monitoring the MIDI messages, it looks like MIDI2LR is sending eight messages roughly every 2 seconds to update the 8 motorized faders on my CM Automation MotorMix controller. Each time it does this, a history entry is created in Lightroom labeled "Multiple Settings".

I also notice that executing a Reset on any of the Basic develop settings like Highlights, Shadows, etc., which are mapped to my MotorMix faders, sets that control to 1 rather than 0. Resetting Exposure sets the value to +0.04 rather than 0. I don't know if that's related or not.

I plan to drop back to an earlier version of MIDI2LR to see if that fixes the problem, as it did not occur with version 2.2.0.0. I will post my findings here.

-David K.

**Comment by dkulik51** [comment]
Interim update: I've tried several earlier versions of MIDI2LR. and all exhibit this same behavior on my system. I wasn't able to try my original working version (2.2.0.0), as it is 32-bit only and doesn't run on MacOS Ventura.

When all this was working for me, I was using Bome Midi Translator Pro to map the MIDI data between the MotorMix and MIDI2LR. In the interest of trying to isolate the source of the "repeat bug," I've created a minimal test configuration by leaving the Bome translator out of the chain and starting with an empty MIDI2LR configuration file. I've found that if I only set CC 0 in MIDI2LR, e.g. mapping it to Basic Tone:Exposure, it will operate correctly without triggering the repeated sends. (It won't update the fader position on the MotorMix in response to slider input in Lightroom, but that is to be expected given the minimal test configuration.) **But** if I set CC 1 or above, any input (including to CC 0) will trigger the repeats. The repeats can be stopped by moving the MotorMix fader mapped to CC 1 to its minimum or maximum position, or moving the mapped slider control in LR, or double-clicking a related setting in LR to reset its value (but that last option is not consistent -- sometimes it works and sometimes it doesn't).

Given the fact that every version of MIDI2LR I was able to test exhibits this behavior, and my latest test configuration rules out issues with my MIDI mapping, this seems like a bug in Lightroom Classic v12.1 (that is still present in v12.2). I'd appreciate any ideas or insights for getting past this, as it's unusable in this state. (For instance, I can't click back in the LR Develop history to see an earlier setting because the next CC repeat will wipe out everything above that point.)

**Reply by mediahope** [reply]
Hi, have you solved the problem?

**Reply by dkulik51** [reply]
Hi -- No; I eventually gave up trying, but I'm still interested in a solution and would be willing to help in any way I can. Thanks for asking.

**Reply by mediahope** [reply]
I tried to solve the problem for a long time and came to the conclusion that you should not have extra midi in the **device manager**.
I had a virtual midi device (_LoopBe1_), I just **disabled its driver**, **removed it from windows startup** and restarted the computer.
Tell me if I could help you! If it didn't work out, then we can try to figure it out.
Good luck to you!

**Comment by dkulik51** [comment]
Well I'm using Mac OS, not Windows, but I'll check to see if I might have
something similar going on. Thanks.

**Reply by mediahope** [reply]
Good luck!

**Comment by hnaef01** [comment]
I have the same Problem.
Win11, Stream Deck+ with Midi-Plugin, Lightroom Classic with Midi2LR. Everything on the latest version
For controlling the DAW (Audio Software) i had to install "LoopMidi", a virtual Midi-Driver, that i use now also for Lightroom.
As long as i send NoteOn / NoteOff with the buttons of the Stream Deck, there is no problem.
As soon as i send CC with the encoder, the values are repeated every about 2 seconds. When i set the value back, it jumps to mysterious values.
What i realised also: When i turn for example the color temperature, i can't set it to zero. It goes from 1 to -1 and back, without 0...

Its difficult to imagine at what point the error happends: Steam Deck Software, Midi PlugIn, virtual Midi driver, Lightroom, Midi2LR... So many players in this game. The OS seams not to be the 'party pooper', as this happens on Mac and Win.

I tried to leave the virtual midi driver, but i have to choose a midi port in the Stream Deck configuration.
Midi2LR isn't visible as a midi port for the controller. Or should it be?

At the moment, i ignore the encoders and configure the buttons with commands of the Ad-Hoc-Development (finest steps: +/-5...).
As i use the Stream Deck also for other things, it's ok. When you use it primary for LR, it's maybe the better way to go for Loupedeck Live or CT...

**Reply by mediahope** [reply]
A little above, I told you that I had the LoopBe1 program installed and problems arose because of it. I think your LoopMidi program works on the same principle.

**Reply by hnaef01** [reply]
Oh - thanks, i didn't realised, that this could be something similar.
But what do you choose as midi port in the Stream Deck software?
And... i need the LoopMidi for my other application, witch i use more often then lightroom...

**Reply by mediahope** [reply]
It is not necessary to completely delete loopMIDI, you can just close or disable it before opening Lightroom (it should work).
I don't have a Stream Deck, I use Arturia Beatstep :)

**Comment by rsjaffe** [comment]
My experience has been that the most common cause of this is having more than one MIDI program in the mix. Trying to pare down the software that is active would help. 

Looking over the advice in [the wiki regarding disabling extra controllers](https://github.com/rsjaffe/MIDI2LR/wiki/MIDI-Controller-Setup#disabling-selected-controllers) may help, as you could have a virtual controller that is causing a feedback loop, since MIDI2LR replies to *all* controllers whenever it sends a message, so one controller could be affecting another if they're both mapped to the same CC or Note number.

**Reply by Masterori** [reply]
Thank you so much!

For all Streamdeck Plus Users.

Using MIDI plug in and LoopMIDI, experiencing this if you follow the above wiki and adjust the output to 0 of your port it solves the repeating signal.

Thank you rsjaffe for not abandoning us!

**Comment by rsjaffe** [comment]
My experience has been that the most common cause of this is having more than one MIDI program in the mix. Trying to pare down the software that is active would help. 

Looking over the advice in [the wiki regarding disabling extra controllers](https://github.com/rsjaffe/MIDI2LR/wiki/MIDI-Controller-Setup#disabling-selected-controllers) may help, as you could have a virtual controller that is causing a feedback loop, since MIDI2LR replies to *all* controllers whenever it sends a message, so one controller could be affecting another if they're both mapped to the same CC or Note number.

**Reply by dkulik51** [reply]
THANK YOU! I had several hardware devices plus input and output items for "IAC Driver Bus 1" listed in my Controllers.xml file, and all were active. None of the hardware devices aside from my MotorMix and MOTU MIDI interface were powered on, but I set their entries to "active=0" anyway. I left only the MotorMix I/O active for my minimal test configuration, and on restarting MIDI2LR, I did not get the repeat bug. So then I started adding back the pieces of my usual working setup, including Bome MIDI Translator. At that point, I disabled the MotorMix entries in Controllers.xml, and enabled the entries for the Bome Translator. All is now back to normal.

Because none of my other hardware devices were powered on, I suspect the culprit here was the IAC Driver entries. I may try to confirm that at some point, but for now I think I will just enjoy using the setup again.

Thanks for coming back to the project, @rsjaffe! I've sent a token of appreciation via your Sponsor link.

David K.

**Comment by LFSaw** [comment]
Just stumbled across this difficulty, it seems that the default mode of MIDI2LR to accept all midi devices is the culprit here:

The IAC driver provides a _bridge_, i.e. everything that goes into its input will show up on its output. Since MIDI2LR picks up all ins and outs by default, it will create an endless loop.
My solution for now is to disable all devices but one bridge input (of a bridge port I called `MIDI2LR IN`) and one bridge output (of a bridge port I called `MIDI2LR OUT`). This results in a stable system.
Unfortunately, clicking on `Rescan MIDI devices` will result in overwriting the `Controllers.xml` file with a new version. Also, restarting the IAC driver will assign new `systemid`s, which then result in new feedback loops that need to be killed by hand each time.

A solution to this would be to not enable all MIDI ports by default but let the user choose manually, which device(s) they want to use for MIDI2LR.

**Comment by Blunkfield** [comment]
Hello everyone,

I am currently configuring the system and have encountered this issue. I have attempted to apply the recommended changes but have been unable to resolve the issue. Has another solution been found?

Best regards.

**Reply by rsjaffe** [reply]
What does your Controllers.xml file look like? What controller are you using?
